metricflow/model/semantics/metric_semantics.py

```diff
@@ -128,17 +128,33 @@ def metric_input_specs_for_metric(self, metric_reference: MetricReference) -> Se
         input_metric_specs: List[MetricSpec] = []
 
         for input_metric in metric.input_metrics:
-            spec_constraint = (
+            input_metric_obj = self.get_metric(input_metric.as_reference)
```

Comment:
nit: maybe we call this `original_metric_obj` to match the `original_metric_constraint` naming below? I guess I should've noticed this the first time....

Reply:
done!